### PR TITLE
bump jenkins and pom version, add incrementals

### PR DESCRIPTION
**Pull request by jetersen**
bumping jenkins version to remove sshd module to avoid cyclic dependencies.

unblocking https://github.com/jenkinsci/bom/pull/1275

@batmat @jglick 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jetersen** [comment]
I need an adult to grant me super powers (write permisson) so I can be allowed to update jenkinsfile.

Not like this changing big things it is just keeping the plugin alive.

**Comment by lemeurherve** [comment]
Replayed a build with manually replaced Jenkins version in the pipeline.

File: pom.xml
Comment (timja) [inline pom.xml:27]:
```suggestion
    <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
```

**Review by timja — APPROVED** [review]

**Comment by lemeurherve** [comment]
> /reviewer batmat

Oh nice, it works even without the `@`, great!

**Comment by timja** [comment]
> > /reviewer batmat
> 
> Oh nice, it works even without the `@`, great!

yup both works for user only :)

**Comment by jetersen** [comment]
need someone with super powers to rebuild, pretty please :)

I guess we need a github-jenkins-comments-op.

Or have Jenkins trust org members :(

**Comment by lemeurherve** [comment]
> need someone with super powers to rebuild, pretty please 😅

Replayed 🙂 

> Or perhaps I can ask for ci.jenkins.io rebuild permission? 👏

Could you open an issue in the help desk for that please?

**Comment by jetersen** [comment]
> Could you open an issue in the help desk for that please?

Done

**Comment by jetersen** [comment]
Can confirm this fixes the issues we are having in bom and https://github.com/jenkinsci/bom/pull/1275

File: pom.xml
Comment (timja) [reply pom.xml:13]: > Or we could go straight to CD, if the maintainer (whoever that is) approves.

@batmat

Comment (jetersen) [reply pom.xml:13]: already created PR for it: https://github.com/jenkins-infra/repository-permissions-updater/pull/2669

File: pom.xml
Comment (basil) [inline pom.xml:56]: It turns out that I also need this fix (the removal of `<type>hpi</type>` for my project, so I wish you a speedy merge and release!

**Comment by basil** [comment]
@jetersen It turns out that in addition to your changes, I also need this change for my work on Maven HPI plugin/PCT:

```xml
diff --git a/pom.xml b/pom.xml
index b0d5525..2b9803c 100644
--- a/pom.xml
+++ b/pom.xml
@@ -67,23 +67,11 @@
     </dependency>
   </dependencies>
 
-  <build>
-    <plugins>
-        <plugin>
-          <groupId>org.apache.maven.plugins</groupId>
-          <artifactId>maven-surefire-plugin</artifactId>
-          <configuration>
-            <classpathDependencyExcludes>
-              <classpathDependencyExcludes>org.jenkins-ci.plugins:beer</classpathDependencyExcludes>
-            </classpathDependencyExcludes>
-          </configuration>
-        </plugin>
-      </plugins>
-    </build>
   <properties>
     <revision>1.5</revision>
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
     <jenkins.version>2.319.3</jenkins.version>
+    <maven.test.dependency.excludes>org.jenkins-ci.plugins:beer</maven.test.dependency.excludes>
   </properties>
 </project>
```

This will allow https://github.com/jenkinsci/maven-hpi-plugin/pull/367 to extend these properties rather than clobbering them.

Any chance you'd be willing to combine efforts and add the above to your PR? I realize that is asking a lot, so no problem if the answer is no - I can wait until after this is merged and then file a separate PR on top of it. But if you're willing to take on these extra low-risk changes, it could save some time.

**Comment by jetersen** [comment]
No problem, let me include it :)

**Pull request merged**
